include/ofi_iov.h

```diff
@@ -74,14 +74,14 @@ ofi_copy_to_iov(const struct iovec *iov, size_t iov_count, uint64_t iov_offset,
 	if (iov_count > 1) {
 		return ofi_copy_iov_buf(iov, iov_count, iov_offset, buf, bufsize,
 					OFI_COPY_BUF_TO_IOV);
-	} else {
-		assert(iov_count == 1);
+	} else if (iov_count == 1) {
 		uint64_t size = ((iov_offset > iov[0].iov_len) ?
 				 0 : MIN(bufsize, iov[0].iov_len - iov_offset));
 
 		memcpy((char *)iov[0].iov_base + iov_offset, buf, size);
 		return size;
 	}
+	return 0;
```

Comment:
Assuming that iov_count == 1 is the common case, we can flip this around:
```
if (iov_count == 1) {
    memcpy(...);
} else {
    ofi_copy_iov_buf(...);
}
```
Note that ofi_copy_iov_buf already handles any iov_count, including 0.

Reply:
Done